File: src/bot/src/services/api/clients/admin_client.py
import requests

from src.common.config import api_link, api_timeout
from src.common.models import UserDto

from src.bot.src.services.api.api_utils import ApiUtils, ApiResponse


class AdminClient:
    __link = f"{api_link}/admin"

    @classmethod
    def accept_user_request(cls, user_id: int, role: str) -> ApiResponse[UserDto]:
        url = f"{cls.__link}/users/{user_id}/accept-role/{role}/"
        response = requests.put(url, timeout=api_timeout)

        return ApiUtils.create_api_response(response, UserDto)

File: src/web_app/src/modules/core/services/api-clients/admin-client.ts
import { httpClient } from '@/modules/core/services/api/http-client';
import type { StatisticsDto, UserRequestDto, Role, ItemsDto } from '@/modules/core/services/api/api.models'
import { type ApiResponse, ApiUtils } from '@/modules/core/services/api/api.utils';

export class AdminClient {
	private static readonly urlBase = 'admin';

	public static async loadRequestsStatistics(): Promise<ApiResponse<StatisticsDto>> {
		const url = `${this.urlBase}/requests-statistics/`;
		const request = httpClient.get(url).json<StatisticsDto>();

		return await ApiUtils.createApiResponse(request);
	}

	public static async loadRequestsByRole(role: Role): Promise<ApiResponse<ItemsDto<UserRequestDto>>> {
		const url = `${this.urlBase}/role-requests/${role}/`;
		const request = httpClient.get(url).json<ItemsDto<UserRequestDto>>();

		return await ApiUtils.createApiResponse(request);
	}

	public static async acceptRoleRequest(requestId: number): Promise<ApiResponse<any>> {
		const url = `${this.urlBase}/role-requests/${requestId}/accept/`;
		const request = httpClient.put(url);

		return await ApiUtils.createApiResponse(request);
	}

	public static async declineRoleRequest(requestId: number): Promise<ApiResponse<any>> {
		const url = `${this.urlBase}/role-requests/${requestId}/decline/`;
		const request = httpClient.put(url);

		return await ApiUtils.createApiResponse(request);
	}
}
